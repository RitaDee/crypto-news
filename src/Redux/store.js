import { configureStore } from '@reduxjs/toolkit';
import coinsReducer from './coins/CoinsSlice';
import coinsDetailsReducer from './details/CoinsDetails';

const store = configureStore({
  reducer: {
    coins: coinsReducer,
    coinsDetails: coinsDetailsReducer,
  },
});

export default store;
