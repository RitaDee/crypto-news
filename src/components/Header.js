import React from 'react';
import '../styles/Header.css';

const Header = () => (
  <div className="headerdiv">
    {/* <img alt="cryptocurrencies" src={allcoins} className="imgheader" /> */}
    <div className="subdiv">
      <h1 className="headerh1">CRYPTO MARKET VALUE</h1>
      <p className="headerh2">100 coins</p>
    </div>
  </div>
);

export default Header;
