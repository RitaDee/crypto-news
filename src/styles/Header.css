.headerdiv {
  display: flex;
  padding-top: 2rem;
  padding-bottom: 2rem;
  padding-left: 0.8rem;
  padding-right: 0.8rem;
  background: rgb(17, 63, 154);
    background: linear-gradient(90deg, rgb(115, 144, 202) 0%, rgba(23, 33, 53, 1) 100%);
    padding: 1rem;
}

.imgheader {
  width: 30%;
}

.subdiv {
  margin: auto;
}

.headerh1 {
  font-weight: 800;
  font-size: 14px;
  line-height: 5px;
  color: aliceblue;
}

.headerh2 {
  font-weight: 700;
  font-size: 12px;
  line-height: 1px;
  color: aliceblue;
}
