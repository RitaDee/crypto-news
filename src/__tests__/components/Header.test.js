import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../Redux/store';
import Header from '../../components/Header';

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Header />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
