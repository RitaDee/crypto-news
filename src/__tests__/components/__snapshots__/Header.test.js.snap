// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="headerdiv"
>
  <div
    className="subdiv"
  >
    <h1
      className="headerh1"
    >
      CRYPTO MARKET VALUE
    </h1>
    <p
      className="headerh2"
    >
      100 coins
    </p>
  </div>
</div>
`;
